src/common/index.ts

```diff
@@ -34,6 +29,16 @@ export class RedditPostData {
   @property author: string;
 }
 
+export class RedditPost {
+
+  constructor(json: any) {
+    Object.assign(this, json as RedditPost);
```

Comment:
Regarding this typeGuard issue: The error is actually justified and the fake typeGuard should be removed. The json contains a plain object, but we want an instance of RedditPostData. The only work around (right now) is to create the instance yourself:

```
this.kind = json.kind;
this.data = new RedditPostData(json.data);
```